### PR TITLE
boards: shields: Add definition of pca63565

### PR DESCRIPTION
**Pull request by nordic-segl**
Define SHIELD pca63565 that is compatible with for example nrf54l15pdk.

### CODE REVIEW

File: boards/shields/pca63565/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	aliases {
+		accel0 = &adxl362;
```
Comment (nordic-bami) [inline boards/shields/pca63565/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:8]: Please add alias for the bme688 sensor.

**Comment by kartben** [comment]
same comment as in #72143 -- I don't understand the need for making a Nordic internal shield available upstream?

**Comment by carlescufi** [comment]
> same comment as in #72143 -- I don't understand the need for making a Nordic internal shield available upstream?

Those are internal shields that can be seen as test fixtures. They allow for additional testing of multiple peripherals, including I2C, I3C, SPI and CAN. While the shields are not available for purchase, they are manufactured and designed to execute vanilla Zephyr tests in our regression test runs. Having them upstream allows us to avoid the maintenance burden of out-of-tree components, and this is not new as we have, to my knowledge, multiple boards (and possibly shields) that are not available for purchase but that you can build (and, if you own it, run) against. As part of this effort we are also contributing additional tests that benefit all Zephyr users (see for example [this PR](https://github.com/zephyrproject-rtos/zephyr/pull/71869)) and that currently use these shields as a fixture.

**Review by erwango — CHANGES_REQUESTED** [review]
Sorry, I tend to agree with @kartben here. To me, upstream, we should only accept code related to hw publicly available, or at the very least with public specification.

I understand the global issue which is about being able to test a specific driver using upstream tests when there is no on board device to enable the test (eg i2c sensor). This is quite a valid requirement but we should find another solution as current proposal doesn't scale. 

If we start accepting internal shields, we'll end up having quite a mess of various funky shields describing ad-hoc hardware for out of tree needs. As a fact we already had numerous PR trying to provide support fantasy shields with one single sensors which were actually simple hardware fixups on top of nucleo boards. If we accept current PR, we'll have hard time rejecting similar proposals.

**Comment by carlescufi** [comment]
> Sorry, I tend to agree with @kartben here. To me, upstream, we should only accept code related to hw publicly available, or at the very least with public specification.

I don't think we've ever enforced anything similar to this, but I understand the concern. Would it help if we made the shields' schematics publicly available?

> I understand the global issue which is about being able to test a specific driver using upstream tests when there is no on board device to enable the test (eg i2c sensor). This is quite a valid requirement but we should find another solution as current proposal doesn't scale.

I agree, but then how do we contribute tests to Zephyr that rely on the hardware being available in some shape or form?
See for example: https://github.com/zephyrproject-rtos/zephyr/pull/71869/files
This is very valuable for the upstream project, because it adds test coverage for existing sensor drivers that can actually be executed on hardware.

> If we start accepting internal shields, we'll end up having quite a mess of various funky shields describing ad-hoc hardware for out of tree needs. As a fact we already had numerous PR trying to provide support fantasy shields with one single sensors which were actually simple hardware fixups on top of nucleo boards. If we accept current PR, we'll have hard time rejecting similar proposals.

I completely understand this point. How can we then contribute the tests without contributing the actual shields in a way that the tests can live upstream and then us, downstream, can build them against the "private" shields?

**Comment by carlescufi** [comment]
@kartben @erwango let's discuss this in the TSC

**Comment by nashif** [comment]
> I agree, but then how do we contribute tests to Zephyr that rely on the hardware being available in some shape or form? See for example: [#71869 (files)](https://github.com/zephyrproject-rtos/zephyr/pull/71869/files) This is very valuable for the upstream project, because it adds test coverage for existing sensor drivers that can actually be executed on hardware.
> 

Such tests will depend on the shield so basically they can't be run upstream unless you have this sheild, correct?

I know of various efforts to make all of this depend on publicly available shields and extensions, @teburd  and @hakehuang  were looking into that. I think something like that is being proposed https://www.mikroe.com/click/click-shields which would allow anyone to re-use the tests.

> I don't think we've ever enforced anything similar to this, but I understand the concern. Would it help if we made the shields' schematics publicly available?

loose enforcement and I there might be a few boards/socs/shields we have in the tree that never made it out of pre-production and probably not available for purchase anywhere. We need to do some audit and cleanup and set the rules a bit more strctly IMO.

I understand the value of adding tests and test shields, but IMO this needs to be coordinated at the project level, if that shield was something public, I do not think there would be any problem. I agree with @erwango  here.

**Comment by teburd** [comment]
If this shield isn't available, its hard to picture this test and shield in the upstream project being useful.

The bme680 test is fine, there's a bme680 click and likely an adafruit breakout (probably others as well) that can be used if the test is written generically, but in that regard I'm surprised we aren't jumping on the i2c_ram test which is simpler, fully excercises the API, and in theory could be used to benchmark throughput and latency as sram should have little effect on this where as a sensor and manipulating its regmap.

We have tons of sensor drivers in the tree, I'd love to see hardware tests for those too in a generic way. But to do that we need to be able to overlay arbitrary sensors and test against those, that means we need the information @yperess is adding in the sensor.yaml to do imho.

I won't block this, but right, do I think its broadly useful compared to testing against a easily purchasable shield or click boards on a click shield? :shrug: not really, it'd be much nicer if this was easily accessible hardware to test with, e.g. i2c sram/fram, spi sram/fram modules, which don't change too much part to part either.

**Comment by carlescufi** [comment]
@teburd @erwango @nashif @kartben so I wonder, if the shield was not available for purchase but its schematics were fully open this would mostly take care fo the concerns, right? Because then anyone can make a similar shield to reuse all the upstream tests.

**Comment by erwango** [comment]
> so I wonder, if the shield was not available for purchase but its schematics were fully open this would mostly take care fo the concerns, right? Because then anyone can make a similar shield to reuse all the upstream tests.

In that case, and out of more documented explicit guidelines, I would remove my nack.
This being said, as mentioned on discord, the problem you're facing is I guess pretty common and we should find a solution to avoid the need to have a shield description in tree only for test purposes as this doesn't really scale.

**Comment by nordic-segl** [comment]
I will try with the downstream:
https://github.com/nrfconnect/sdk-nrf/pull/15253

**Pull request closed without merging**